### PR TITLE
Improve custom server typescript example

### PR DESCRIPTION
**Pull request by TheRobBrennan**
This PR enhances the TypeScript example with the following key enhancements:
+ The current TypeScript server example transpiles the source `server/server.ts` into `production-server/server.js`. This will fail if deployed to Zeit Now. This update stores the generated file in `.next/production-server`
+ Updated TypeScript to `^2.8.3`
+ Updated React to `^16.3.2`
+ Updated react-dom to `^16.3.2`

Babel libraries have also been added so that the user can successfully run `npm install` and `npm run dev` successfully.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "next",
-  "version": "6.0.3-canary.1",
+  "version": "6.0.3",
```
Comment (timneutkens) [inline package.json]: 🤔 Sounds like you created from the wrong branch.

Comment (TheRobBrennan) [reply package.json]: @timneutkens Hmmm I forked the repo, cloned that to my local machine...did a get fetch upstream to make sure my canary was the latest, and then branched off that. Did I miss a step somewhere?

Comment (timneutkens) [reply package.json]: @TheRobBrennan try doing `git pull origin canary` where origin is zeit/next.js. Besides this I want to merge the PR 😄

File: examples/custom-server-typescript/package.json [hunk 2]
```diff
@@ -2,14 +2,17 @@
...
     "build": "next build && tsc --project tsconfig.server.json",
-    "start": "NODE_ENV=production node production-server/index.js"
+    "start": "NODE_ENV=production node .next/production-server/index.js"
   },
   "dependencies": {
-    "next": "latest",
-    "react": "^16.2.0",
-    "react-dom": "^16.2.0",
+    "@babel/core": "^7.0.0-beta.47",
     "@zeit/next-typescript": "0.1.1",
```
Comment (timneutkens) [inline examples/custom-server-typescript/package.json]: I think you can upgrade this to use the babel solution for typescript.

**Comment by TheRobBrennan** [comment]
Gah - I cannot get this branch to cooperate. Closing this PR; opened PR https://github.com/zeit/next.js/pull/4470/ instead

**Pull request closed without merging**